lib/shellable.ts

```diff
@@ -186,6 +186,55 @@ export interface AssumeRole {
    * underscores or  any  of  the  following characters: =,.@:/-
    */
   externalId?: string;
+
+  /**
+   * When a profie name is configured, an assumed role configuration will be created
+   * in the shared aws configuration file (~/.aws/config). This is in contrary of simply invoking
+   * an `sts assume-role` command that creates a session with a fixed expiry date.
+   *
+   * Using a profile will delegate credential refreshing to the SDK/CLI.
+   * This is needed to support long running sessions that needs sessions that are longer than
+   * the session duration that can be configured with a `sts assume-role`.
+   *
+   * When a profile name is specified, you must also configure one of `sourceProfile` or `credentialSource`.
+   *
+   * @see https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-files.html
+   */
+  profileName?: string,
+
+  /**
+   * A profile name to be used as the credential with which to assume the role.
+   * Must exist in either the credential or the config file.
+   * Either this or `credentialSource` is required when using `profileName`
+   *
+   * @see https://docs.aws.amazon.com/credref/latest/refdocs/setting-global-source_profile.html
+   */
+  sourceProfile?: string,
```

Comment:
There is no point in this one, really. Yes it's for full capabilities, but in the context of what a Shellable is (just a run-once CodeBuild project) it doesn't make any sense. Who's gonna be writing that other profile entry?

Just drop this.

Reply:
I was thinking about if the shellable is executed in some environment where the shared files actually exist.
Agree its way pre-mature for that.

I'll remove.